Core/Src/main.c

```diff
@@ -87,17 +92,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
+
+  // Starts PWM
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t DataReceived[DATASIZE] = {0x00, 0x80, 0x00};
+  uint8_t DataTransmitted[DATASIZE] = {0x00};
+
+
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // Sets CS line to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // Uses appropriate SPI protocol to transmit and receive ADC values
+	  HAL_SPI_TransmitReceive(&hspi1, DataTransmitted, DataReceived, sizeof(DataTransmitted), 24);
     /* USER CODE END WHILE */
 
+	  // Delay between processes to ensure that MCU doesnt overload
+	  HAL_Delay(10);
```

Comment:
Though I know you set a delay here for a good reason. But the delay here is going to occupy this SPI port for communicating to ADC chip for 10ms. This is a really long time for both the mcu and the adc chip. And if this SPI is connected to multiple sensors for instance, this also means in this 10ms it can't talk to other devices. So for good practice, we usually keep the delay between the CS line on/off bound as short as possible.